tools/src/bin/output-file.rs

```diff
@@ -335,22 +335,58 @@ fn main() {
                     accel_key: None,
                     copyable: false,
                 });
-                if let Some(ref hg_root) = tree_config.paths.hg_root {
+
+                let gh_log_link = tree_config
+                    .paths
+                    .github_repo
+                    .as_ref()
+                    .map(|gh_root| format!("{}/commits/HEAD/{}", gh_root, encoded_path));
```

Comment:
Have you seen any docs that say this will end up defaulting to the default branch?  I do see at https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits--parameters that the Query parameters docs for `sha` says:
> SHA or branch to start listing commits from. Default: the repository’s default branch (usually main).

So maybe this is good enough?

(It does seem hopefully unlikely that if someone pushed a branch other than the default branch that github would be favor that commit over the default branch head.)

Reply:
I've not found any docs for that no, but it works, at least currently. I have no idea what promises GitHub makes for its user-facing URLs.
Note that the documentation you linked to is for api.github.com, which is properly documented and versioned, but not a good place to redirect humans to :wink:

Reply:
Ha, yeah, I just meant to hand-wave that presumably the UI is operating on top of the same logic as the API so ideally it's stable.